components/metrics/lib.rs

```diff
@@ -128,12 +134,20 @@ impl InteractiveWindow {
     //   restart: there was a task > 50ms
     //   not all documents are interactive
     pub fn start_window(&mut self) {
-        self.start = precise_time_ns();
+        self.start = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap_or_default()
+            .as_nanos() as u64;
     }
 
     /// check if 10s has elapsed since start
     pub fn needs_check(&self) -> bool {
-        precise_time_ns() - self.start >= INTERACTIVE_WINDOW_SECONDS_IN_NS
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap_or_default()
+            .as_nanos() as u64 -
+            self.start >=
```

Comment:
I think you could store `self.start` as a `SystemTime` and also `INTERACTIVE_WINDOW_DURATION` as a `Duration` here and then simply do:

```rust
SystemTime::now().duration_since(self.start).unwrap_or_default() >= INTERACTIVE_WINDOW_DURATION
```

which would simplify a lot of the code in this file.

Reply:
I needed this is much better, thanks